### DISCUSSION TITLE
Is nested populated still working as described in the docs?

### DISCUSSION THREAD

**Discussion by co-sic** [mikro-orm/mikro-orm#2771] [Q&A] [answered]
https://mikro-orm.io/docs/nested-populate
There is an example for populating a colletion and then populate fields on the collection:

`const tags = await em.findAll(BookTag, { populate: ['books.publisher.tests', 'books.author'] });`

i used this with v4, but with v5.0.1 and the strictly typed populate it is not working anymore. I only can select the properties of the Collection class, like add, count, contains, etc.

**Comment by B4nan** [comment, marked as answer]
yes, it works just fine. tests are full of this.

> I only can select the properties of the Collection class, like add, count, contains, etc.

sounds like wrong entity definition, or some weird problem with type inference (like the `Collection` type was not detected)

**Reply by B4nan** [reply]
do you have problems with `findAll` specifically, or in general with repositories? there is no `em.findAll` as in your example, that is only a repository method

we have extensive tests for EM api, but not so much for repositories (as they basically just forward the calls to EM). maybe it is something in the repository types? did a quick test but it all looks good to me

**Reply by co-sic** [reply]
The example was taken out of the docs i linked above, not from my code. It was happening in general for me after upgrading to mikro-orm v5. I found the problem now, it was the second generic from Collection. I defined it as `books = new Collection<Book, Author>(this)` on and Author entity. After removing the second generic the populate typing was working again. Thanks for the pointers. So should you never use the second generic? Or is there another use case for this?

**Reply by B4nan** [reply]
Interesting, it shouldnt matter, will take a look.

**Reply by B4nan** [reply]
https://github.com/mikro-orm/mikro-orm/commit/e39ef5b9acdb5f052d9e4f0d3d6c04af5f43eefb
